src/sentry/search/events/datasets/discover.py

```diff
@@ -1434,6 +1432,52 @@ def _resolve_user_misery_function(self, args: Mapping[str, str], alias: str) ->
             alias,
         )
 
+    def _resolve_count_if(self, args: Mapping[str, str], alias: str) -> SelectType:
+        condition = args["normalized_condition"]
+        is_array_field = args["is_array_field"]
+        if not is_array_field:
```

Comment:
nit: Could we flip this if condition, or add an `else`, cause this makes L1460 reads funky otherwise